play/json-flattened-view-gen/src/main.rs

```diff
@@ -0,0 +1,110 @@
+use serde_json::value::Value;
+use structopt::StructOpt;
+
+#[derive(Debug)]
+enum PostgresType {
+    Text,
+    Other,
+}
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Sample json was not an serde_json::Value::Object.")]
+    NotAnObject,
+}
+
+fn flatten_object(map: Value) -> Result<Vec<(String, PostgresType)>, Error> {
+    if let Value::Object(m) = map {
+        let mut flattened = Vec::new();
+        for (key, val) in m.into_iter() {
+            flattened.extend(flatten_json(key, val));
+        }
+        Ok(flattened)
+    } else {
+        Err(Error::NotAnObject)
+    }
+}
+
+fn flatten_json(key: String, value: Value) -> Vec<(String, PostgresType)> {
```

Comment:
`flatten_object` and `flatten_json` could use top-level comments; it's not very clear just from the signature what they do.

Reply:
It doesn't matter since this code is off the critical path, but if you want, you could save a lot of allocations by having this take a `&mut Vec<(String, PostgresType)>` argument and write to it directly, rather than always creating a new vector and then extending the old vector in the next stack frame up.

Reply:
Commented and switched the `&mut Vec`. I also renamed `flatten_json` to `flatten_single_value` to hopefully make it more clear.